src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+                ),
+                data={
+                    "web.conf": ol_app_k8s_config.project_root.joinpath(
+                        "files/web.conf"
+                    ).read_text(),
+                },
+                opts=resource_options,
+            )
+
+        # Build a list of not-sensitive env vars for the deployment config
+        application_deployment_env_vars = []
+        for k, v in (self.ol_app_k8s_config.application_config).items():
+            application_deployment_env_vars.append(
+                kubernetes.core.v1.EnvVarArgs(
+                    name=k,
+                    value=v,
+                )
+            )
+        application_deployment_env_vars.append(
+            kubernetes.core.v1.EnvVarArgs(name="PORT", value=str(DEFAULT_UWSGI_PORT))
+        )
+
+        # Build a list of sensitive env vars for the deployment config via envFrom
+        application_deployment_envfrom = [
+            # Database creds
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.db_creds_secret_name,
+                ),
+            ),
+            # Redis Configuration
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.redis_creds_secret_name,
+                ),
+            ),
+            # static secrets from secrets-application/secrets
+            kubernetes.core.v1.EnvFromSourceArgs(
+                secret_ref=kubernetes.core.v1.SecretEnvSourceArgs(
+                    name=self.ol_app_k8s_config.static_secrets_name,
+                ),
+            ),
+        ]
+
+        init_containers = [
+            # Run database migrations at startup
+            kubernetes.core.v1.ContainerArgs(
+                name="migrate",
+                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                command=["python3", "manage.py", "migrate", "--noinput"],
+                image_pull_policy="IfNotPresent",
+                env=application_deployment_env_vars,
+                env_from=application_deployment_envfrom,
+            ),
+            kubernetes.core.v1.ContainerArgs(
+                name="collectstatic",
+                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                command=["python3", "manage.py", "collectstatic", "--noinput"],
+                image_pull_policy="IfNotPresent",
+                env=application_deployment_env_vars,
+                env_from=application_deployment_envfrom,
+                volume_mounts=[
+                    kubernetes.core.v1.VolumeMountArgs(
+                        name="staticfiles",
+                        mount_path="/src/staticfiles",
+                    ),
+                ],
+            ),
+        ] + [
```

Comment:
We should think about how we can make this a bit more flexible. This is a pattern that will be common for Django apps, but if we want/need to adapt to other application patterns we might want a different set of init containers.